tests/api/clever/test_clever.py

```diff
@@ -67,10 +68,13 @@ def test_external_type_from_clever_grade(self):
             assert external_type_from_clever_grade(none_grade) is None
 
 
-class TestCleverAuthenticationAPI(DatabaseTest):
-    def setup_method(self):
-        super().setup_method()
-        self.api = MockAPI(self._default_library, self.mock_integration)
+class CleverAuthenticationFixture:
+    db: DatabaseTransactionFixture
+    api: MockAPI
+
+    def __init__(self, db: DatabaseTransactionFixture):
+        self.db = db
+        self.api = MockAPI(db.default_library(), self.mock_integration)
         os.environ["AUTOINITIALIZE"] = "False"
         from api.app import app
```

Comment:
I think this carries the risk of polluting the environment for further test runs, unless I'm wrong.

The code should probably set `os.environmen["AUTOINITIALIZE"]` and then restore it in a `close` method. This will require making the fixture method into one that yields a value and then performs the close operation afterwards. Take a look at, for example, the end-to-end search fixture:

https://github.com/ThePalaceProject/circulation/blob/main/tests/fixtures/search.py#L278

Reply:
make sense 👍 https://github.com/ThePalaceProject/circulation/pull/825/commits/a537f274feadb80e1fa9336a1f313e7f2a35b5a6